src/lsptoolshost/roslynLanguageServer.ts

```diff
@@ -349,8 +365,19 @@ export class RoslynLanguageServer {
             } else {
                 // Auto open if there is just one solution target; if there's more the one we'll just let the user pick with the picker.
                 const solutionUris = await vscode.workspace.findFiles('**/*.sln', '**/node_modules/**', 2);
-                if (solutionUris && solutionUris.length === 1) {
-                    this.openSolution(solutionUris[0]);
+                if (solutionUris) {
+                    if (solutionUris.length === 1) {
+                        this.openSolution(solutionUris[0]);
+                    } else if (solutionUris.length === 0) {
+                        // We have no solutions, so we'll enumerate what project files we have and just use those.
+                        const projectUris = await vscode.workspace.findFiles(
+                            '**/*.csproj',
+                            '**/node_modules/**',
+                            options.omnisharpOptions.maxProjectResults
```

Comment:
This is an omnisharp option that I'm consuming here even on the non-omnisharp side, in case there is a user out there where this limit is significant. Having this not be a dotnet option is a bit of a violation of naming though, but I'm not sure I want to create a new option and setup the migration at this moment until we have a chance to spec it a bit more. Put another way, I'm OK with this bug at the moment.

Reply:
Fine with as-is - lets have an issue to track.  But it should be pretty straightforward to migrate the option name if you wanted.  We just put the O# name as the fallback, then add a value to https://github.com/dotnet/vscode-csharp/blob/main/src/shared/migrateOptions.ts

Reply:
To echo something I mentioned to @dibarbet privately: making the fallback work is easy, but the option is a bit funky in this case since the filtering isn't exactly deterministic in what it will filter out: it prevents certain large repositories from causing pathological problems, but it's not ideal.